### PR TITLE
[REF] l10n_au_keypay: remove au dependency

### PR DESCRIPTION
**Pull request by vin-odoo**
Keypay, now called Employment Hero, is not limited to Australia but also supports Singapore, Malaysia,...

This task will:
- remove dependencies to Australia, to make it usable with other localisation.
- Rename the module and all references to keypay and au in order to keep the module clean for future use.
- Implement better error messages when the api returns one.
- Render the api key, and url company specific to support using two different API on two separate companies.

Task id # 2925767

### CODE REVIEW

**Review by chiaraprattico — COMMENTED** [review]
Hi @vin-odoo thanks for the PR, I rephrased some sentences and I updated one of the screenshots.

**Comment by vin-odoo** [comment]
@odoo/be-doc-review Hi team! The enterprise PR has been r+ed, do you think you'll have some time to check before tomorrow? Thanks 😄

**Comment by jcs-odoo** [comment]
> @odoo/be-doc-review Hi team! The enterprise PR has been r+ed, do you think you'll have some time to check before tomorrow? Thanks 😄

okay :)

File: content/applications/finance/fiscal_localizations/australia.rst
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/australia.rst:1]: I suppose we need to remove lines 38-40
```
   * - :guilabel:`Employment Hero Australian Payroll`
     - `l10n_au_keypay`
     - Synchronises all pay runs from Employment Hero with Odoo’s journal entries.
```

File: content/applications/finance/fiscal_localizations/new_zealand.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+===========
+New Zealand
+===========
+
+.. _new-zealand/employment-hero:
+
+Employment Hero New Zealand Payroll
+===================================
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/new_zealand.rst:8]:
```suggestion
Employment Hero Payroll
=======================
```

File: content/applications/finance/fiscal_localizations/new_zealand.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+If your business is already up and running with `Employment Hero <https://employmenthero.com/>`_,
+you can use our connector as an alternative payroll solution.
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/new_zealand.rst:11]: I'd rather link our employment hero doc here and remove the seealso section, wdyt?

```suggestion
If your business is already up and running with :doc:`Employment Hero <employment_hero>`, you can
use our connector as an alternative payroll solution.
```

**Review by jcs-odoo — APPROVED** [review]
I'll push the changes I requested in a second.

**Comment by jcs-odoo** [comment]
Hey @vin-odoo 

I applied the changes I suggested in my review, so you can merge like this if they're okay for you.
Have a good evening

@robodoo delegate+

**Comment by vin-odoo** [comment]
@jcs-odoo Thanks a lot, the changes looks great!

@robodoo r+

**Pull request closed without merging**